src/main/java/org/apache/ibatis/builder/xml/XMLMapperBuilder.java

```diff
@@ -117,7 +117,7 @@ private void configurationElement(XNode context) {
       sqlElement(context.evalNodes("/mapper/sql"));
       buildStatementFromContext(context.evalNodes("select|insert|update|delete"));
     } catch (Exception e) {
-      throw new BuilderException("Error parsing Mapper XML. Cause: " + e, e);
+      throw new BuilderException("Error parsing Mapper XML." + ErrorContext.instance() + " Cause: " + e, e);
```

Comment:
It's better not to use `ErrorContext`.
The mapper file location is in `this.resource`, so please use it to build the message.

Reply:
It is better to use `resource` than `ErrorContext` In this case,because nobody knows how much infomation the context contains .Also it is the weakness to use context(For example
`Chain of Responsibility` )
But I am curious about the code design of `ErrorContext ` .I'm puzzled about using `ErrorContext `,I can find putting the infomation in it,but I cannot find where to get the infomation
Thank You.

Reply:
It is used in `ExceptionFactory#wrapException()`, I think.

Reply:
Thank you for your prompt answer